crates/common/src/state.rs

```diff
@@ -229,20 +231,21 @@ pub trait State {
         &mut self,
         sudt_id: u32,
         address: &RegistryAddress,
-        amount: u128,
+        amount: impl Into<U256>,
```

Comment:
Generics like this if monomorphized many times can increase increase compile time and binary size.

I suggest that you just use `amount: U256` here and let the caller call `.into()`.